Objects/typeobject.c

```diff
@@ -6599,6 +6599,12 @@ object_set_class(PyObject *self, PyObject *value, void *closure)
     }
 
     PyTypeObject *oldto = Py_TYPE(self);
+#ifdef Py_GIL_DISABLED
+    PyInterpreterState *interp = PyInterpreterState_Get();
```

Comment:
nitpick, you can use the faster static inline function:

```suggestion
    PyInterpreterState *interp = _PyInterpreterState_GET();
```